src/world/Objects/GameObject.cpp

```diff
@@ -254,21 +261,151 @@ void GameObject::setAnimationProgress(uint8_t progress)
+{
+    if (getCreatedByGuid() != 0)
+        return getWorldMapUnit(getCreatedByGuid());
+
+    return nullptr;
+}
+
 Player* GameObject::getPlayerOwner()
 {
-    if (m_summoner != nullptr && m_summoner->isPlayer())
-        return dynamic_cast<Player*>(m_summoner);
+    if (getCreatedByGuid() != 0)
+        return getWorldMapPlayer(getCreatedByGuid());
 
     return nullptr;
 }
-// MIT End
 
-GameObjectProperties const* GameObject::GetGameObjectProperties() const
+bool GameObject::loadFromDB(MySQLStructure::GameobjectSpawn* spawn, WorldMap* map, bool addToWorld)
 {
-    return gameobject_properties;
+    if (!spawn)
+    {
+        sLogger.failure("Gameobject (spawnId: %u) not found in table gameobject_spawns, cant load.");
+        return false;
+    }
+
+    if (!map || !map->getBaseMap())
+    {
+        sLogger.failure("Gameobject (spawnId: %u) invalid WorldMap or base data Invalid, cant load.");
+        return false;
+    }
+
+    uint32_t entry = spawn->entry;
+    GameObject_State gameobjectState = spawn->state;
+
+    m_spawnId = spawn->id;
+    if (!create(entry, map, spawn->phase, spawn->spawnPoint, spawn->rotation, spawn->state))
+        return false;
+
+    if (spawn->spawntimesecs >= 0)
+    {
+        m_spawnedByDefault = true;
+
+        if (!GetGameObjectProperties()->getDespawnPossibility() && !GetGameObjectProperties()->isDespawnAtAction())
+        {
+            setFlags(GO_FLAG_NEVER_DESPAWN);
+            m_respawnDelayTime = 0;
+            m_respawnTime = 0;
+        }
+        else
+        {
+            m_respawnDelayTime = spawn->spawntimesecs;
+            m_respawnTime = map->getGORespawnTime(spawn->id);
+
+            // ready to respawn
+            if (m_respawnTime && m_respawnTime <= Util::getTimeNow())
+            {
+                m_respawnTime = 0;
+                map->removeRespawnTime(SPAWN_TYPE_GAMEOBJECT, spawn->id);
+            }
+        }
+    }
+    else
+    {
+        m_spawnedByDefault = false;
+        m_respawnDelayTime = -spawn->spawntimesecs;
+        m_respawnTime = 0;
+    }
+
+    m_spawn = spawn;
+
+    // add to insert Pool
+    if (addToWorld)
+        map->AddObject(this);
+
+    return true;
 }
 
-bool GameObject::CreateFromProto(uint32 entry, uint32 mapid, float x, float y, float z, float ang, float r0, float r1, float r2, float r3, uint32 overrides)
+void GameObject::deleteFromDB()
+{
+    if (m_spawn != nullptr)
+    {
+        WorldDatabase.Execute("DELETE FROM gameobject_spawns WHERE id = %u AND min_build <= %u AND max_build >= %u ", m_spawn->id, VERSION_STRING, VERSION_STRING);
+        WorldDatabase.Execute("DELETE FROM gameobject_spawns_extra WHERE id = %u AND min_build <= %u AND max_build >= %u ", m_spawn->id, VERSION_STRING, VERSION_STRING);
+        WorldDatabase.Execute("DELETE FROM gameobject_spawns_overrides WHERE id = %u AND min_build <= %u AND max_build >= %u ", m_spawn->id, VERSION_STRING, VERSION_STRING);
+    }
+}
+
+void GameObject::saveToDB()
```

Comment:
That function is kind of odd.
If it is a new spawn, save the data with the current Version. If the spawn is already in gameobject_spawns just update the data and keep min/max_build as it was...